src/dmd/dtoh.d

```diff
@@ -204,36 +233,13 @@ void genCppHdrFiles(ref Modules ms)
     buf.writeByte('\n');
     buf.writestring("#include <assert.h>\n");
     buf.writestring("#include <stddef.h>\n");
+    buf.writestring("#include <stdint.h>\n");
     buf.writestring("#include <stdio.h>\n");
     buf.writestring("#include <string.h>\n");
     buf.writeByte('\n');
-    buf.writestring("#define _d_void void\n");
-    buf.writestring("#define _d_bool bool\n");
-    buf.writestring("#define _d_byte signed char\n");
-    buf.writestring("#define _d_ubyte unsigned char\n");
-    buf.writestring("#define _d_short short\n");
-    buf.writestring("#define _d_ushort unsigned short\n");
-    buf.writestring("#define _d_int int\n");
-    buf.writestring("#define _d_uint unsigned\n");
-    if (global.params.isLP64)
-    {
-        buf.writestring("#define _d_long long\n");
-        buf.writestring("#define _d_ulong unsigned long\n");
-    }
-    else
-    {
-        buf.writestring("#define _d_long long long\n");
-        buf.writestring("#define _d_ulong unsigned long long\n");
-    }
-    buf.writestring("#define _d_float float\n");
-    buf.writestring("#define _d_double double\n");
-    buf.writestring("#define _d_real long double\n");
-    buf.writestring("#define _d_char char\n");
-    buf.writestring("#define _d_wchar wchar_t\n");
-    buf.writestring("#define _d_dchar unsigned\n");
-    buf.writestring("typedef _d_long d_int64;\n");
-    buf.writestring("\n");
-    buf.writestring("#define _d_null NULL\n");
+    buf.writestring("#if !defined(_d_real)\n");
+    buf.writestring("# define _d_real long double\n");
```

Comment:
If you removed the others, why keep this around?

Perhaps per-target mappings can be maintained as separate enums that can be used to instantiate the translator. Though just thinking out loud.

Reply:
I figured it may be possible that some projects might want to define real as double... and it's 2 words, so the code the includes this header can't redefine it.

Reply:
Nah, if you're doing that, you'll already be compiling with `-mlong-double-64`, so both C and D abis match up.

Reply:
So you recommend removing this macro? I support this if it seems reasonable and remains portable.

Reply:
The macro can be removed.